### DISCUSSION TITLE
Docker upgrade PR title customization

### DISCUSSION THREAD

**Discussion by Stephan202** [renovatebot/renovate#13192] [Request Help] [answered]
We have a configuration containing the following:

```js
module.exports = { 
  ...
  commitMessageAction: "Upgrade",
  commitMessageTopic: "{{depName}}",
  commitMessageExtra: "{{currentVersion}} -> {{newVersion}}",
  ...
```

For Maven dependencies this works fine, producing PR titles such as `Upgrade com.bugsnag:bugsnag 3.6.2 -> 3.6.3`.

However, for Docker image upgrades it appears that this configuration is ignored, and a [default](https://github.com/renovatebot/renovate/blob/25d3d626c91237fb7eddbc638a383399bdf07755/lib/datasource/docker/index.ts#L35-L60) title such as `Upgrade mongo Docker tag to v5` is generated instead. We tried to fix this by extending the package rules, as follows:
```js
  ...
  packageRules: [
    ...
    {
      matchDatasources: ["docker"],
      commitMessageTopic: "{{{depName}}}",
      commitMessageExtra: "{{{currentVersion}}} -> {{{newVersion}}}",
    },
    ...
```
But this results in PR titles of the form `Upgrade -> `, i.e. the setting is respected, but variable interpolation fails. We get the same result if we use two rather than three curly braces. Other variables such as `replaceString`, `currentValue` and `newValue` aren't interpolated either.

We wonder: are we "holding it wrong", or is there some bug here? We observed this issue with Renovate 29.23.3. Upgrading to Renovate 31.6.0 did not help.

**Comment by rarkins** [comment]
It's possible that these docker datasource defaults are taking precedence: https://github.com/renovatebot/renovate/blob/3b91162ecadbddad653981c8b588978d317bd197/lib/datasource/docker/index.ts#L35-L60

If you can't work out how to achieve what you need after that, please create a [minimal public reproduction](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) for us to look at.

**Comment by Stephan202** [comment]
@rarkins thanks for the rapid response. I just took over responsibility for a Renovate deployment and so far have been very impressed with the community support.

... and the fact that we have a custom Renovate deployment turns out to be related to the described issue: we have a customized setup which preprocesses/generates `config.js` using Mustache. This explains why the `packageRules` approach failed: I didn't escape the curly braces, which caused the preprocessor to interpolate the variables, making them invisible to Renovate. [This](https://stackoverflow.com/a/13948304/74939) resolved that.

Strictly speaking this unblocks me, but it's an open question why the top-level `commitMessageTopic` and `commitMessageExtra` don't work (that's existing configuration I inherited, which was already exempted from interpolation by Mustache, so the root cause there is something else). I'll look into creating a minimal public reproduction, though it might take a while as I'm still getting up to speed with Renovate.

Thanks again!

**Comment by Stephan202** [comment, marked as answer]
@rarkins alright, I managed to create a minimal public reproduction; please take a look at [this repository](https://github.com/PicnicSupermarket/renovate-discussion-13192). I also added a Helm data source, as it seem to exhibit the same issue.

**Reply by rarkins** [reply]
At the moment this looks to be the same as https://github.com/renovatebot/renovate/issues/7281

BTW I think that your config will produce undesirable results if you have digest updates in Docker, which is something you should do in future because Docker tags are immutable. In the case of digest update site current and new values are the same

**Reply by Stephan202** [reply]
> At the moment this looks to be the same as #7281

Thanks, I'll subscribe to that issue. :+1:

> BTW I think that your config will produce undesirable results if you have digest updates in Docker, which is something you should do in future because Docker tags are immutable. In the case of digest update site current and new values are the same

Yep, I realized the same last night :sweat_smile: (indeed we're slowly adopting digests everywhere). Another entry on the list of things to improve before enabling Dockerfile/Docker Compose update PRs company-wide. :)

Thanks again!

**Reply by Stephan202** [reply]
Hmm, your reference to #7281 should be considered the answer to the original question, but as it's part of a reply I cannot click _Mark as answer_ on it. I'll instead mark the top-level message as such; let me know if you prefer an alternative approach.

**Reply by Stephan202** [reply]
(Yeah, now it says _Answered by Stephan202_, which isn't strictly true :man_shrugging:.)
